src/components/KTreeList/KTreeList.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <div class="k-tree-list">
+    <KTreeDraggable
+      :items="internalList"
+      :disable-drag="disableDrag"
+      :max-level="maxLevel"
+      @change="handleChangeEvent"
+      @child-change="handleChildChangeEvent"
+      @selected="handleSelection"
+    >
+      <template #item-icon="{ item }">
+        <slot
+          :item="item"
+          name="item-icon"
+        />
+      </template>
+      <template #item-label="{ item }">
+        <slot
+          :item="item"
+          name="item-label"
+        />
+      </template>
+    </KTreeDraggable>
+  </div>
+</template>
+
+<script lang="ts" setup>
+/**
+ *  Note: if TS errors appear on the
+ *  <template #[itemIcon]="slotProps"> or
+ *  <template #[itemLabel]="slotProps">
+ * lines switch to `v-slot:` notation and Save to let linter clear -->
+ */
+import { computed, ref, watch, onMounted, PropType } from 'vue'
+import KTreeDraggable from '@/components/KTreeList/KTreeDraggable.vue'
+import { TreeListItem } from '@/components/KTreeList/KTreeItem.vue'
+
+export interface ChangeEvent {
+  items: TreeListItem[]
+}
+
+export interface ChildChangeEvent {
+  parent: string,
+  children: TreeListItem[]
+}
+
+const props = defineProps({
+  modelValue: {
+    type: Array as PropType<TreeListItem[]>,
+    default: null,
+    validator: (items: TreeListItem[]) => !items.length || items.every(i => i.name !== undefined && i.id !== undefined),
+  },
+  items: {
+    type: Array as PropType<TreeListItem[]>,
+    default: null,
+    validator: (items: TreeListItem[]) => !items.length || items.every(i => i.name !== undefined && i.id !== undefined),
```

Comment:
A couple issues:

In the validator (here on `modelValue` as well as the `items` validator), you also need to loop through all items and ensure all `id` properties are unique. 

Additionally, this validator doesn't account for validating children recursively, meaning the L1 items may pass validation, but all children (infinite levels) would never be evaluated. 

Since you're using the same validation logic on both props, I would extract the logic into a function to call within the validator that returns a boolean. Then you can use the same function wherever the validator is needed.